universal-login-relayer/src/core/services/validators/EmailConfirmationValidator.ts

```diff
@@ -0,0 +1,19 @@
+import moment from 'moment';
+import {ensureNotFalsy, ensure} from '@unilogin/commons';
+import {EmailConfirmationsStore} from '../../../integration/sql/services/EmailConfirmationsStore';
+import {DuplicatedEmail, EmailNotFound, InvalidCode, CodeExpired} from '../../utils/errors';
+
+export class EmailConfirmationValidator {
+  constructor(
+    private emailConfirmationsStore: EmailConfirmationsStore,
+    private codeDurationInMinutes: number = 5,
+  ) {}
+
+  async validate(email: string, code: string) {
+    const emailConfirmation = await this.emailConfirmationsStore.get(email);
+    ensureNotFalsy(emailConfirmation, EmailNotFound, email);
+    ensure(!emailConfirmation.isConfirmed, DuplicatedEmail, email);
```

Comment:
In this case, we should have `DuplicatedConfirmation` instead of `DuplicatedEmail`